test/e2e/instrumentation/tracing/apiserver.go

```diff
@@ -0,0 +1,201 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tracing
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/component-base/traces"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	e2essh "k8s.io/kubernetes/test/e2e/framework/ssh"
+	instrumentation "k8s.io/kubernetes/test/e2e/instrumentation/common"
+)
+
+const (
+	podName       = "otel-collector"
+	containerName = "collector"
+	httpTarget    = "/api/v1/nodes"
+)
+
+// The API Server Tracing test ensures that an opentelemetry collector can
+// collect traces from the API Server, and that context is correctly propagated
```

Comment:
maybe I'm missing it, but I'm not seeing how the apiserver and collector pod are wired together… where is that happening?

Reply:
When tracing is enabled, the apiserver sends traces to the default port for OTLP.  When we schedule a collector on the control-plane node, it listens on the default port for OTLP, and the apiserver will connect to it and send traces.  In the collector config below, you can see the collector has the OTLP grpc "receiver".

Reply:
huh… so does this only work on a single master node where kube-apiserver is running with hostNetwork: true?

Reply:
Yes.  We have a [plan](https://github.com/kubernetes/kubernetes/pull/103234#discussion_r659854190) for how to improve the test to work with multiple masters, but right now it assumes the apiserver is running on host network.

Reply:
yeah, network namespace isolation for the core components was out of scope for rootless.